src/main/java/net/sourcewriters/minecraft/vcompat/logging/BukkitLogger.java

```diff
@@ -0,0 +1,227 @@
+import com.syntaxphoenix.syntaxapi.utils.java.Times;
+import net.sourcewriters.minecraft.vcompat.reflection.reflect.ReflectionProvider;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.craftbukkit.Main;
+import org.bukkit.craftbukkit.libs.jline.console.ConsoleReader;
+import org.bukkit.plugin.Plugin;
+import org.fusesource.jansi.Ansi;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.function.BiConsumer;
+
+public class BukkitLogger implements ILogger {
+
+    private static final String ANSI_RESET = Ansi.ansi().reset().toString();
+
+    private final boolean ansiSupported, jLine;
+    private final LogTypeMap typeMap = new LogTypeMap();
+    private final Writer writer;
+    private final ConsoleReader reader;
+    private final String plugin;
+
+    private LoggerState state;
+    private String format = "[%date%/%time%] [Server thread/%plugin% | %type%]: %message%";
+
+    public BukkitLogger(Plugin plugin) {
+        this(plugin.getName());
+    }
+
+    public BukkitLogger(String plugin) {
+        this.plugin = plugin;
+        // TODO: Don't use reflections
+        Object mcServer = ReflectionProvider.DEFAULT.createCBReflect("CraftServer", "CraftServer")
+                .searchField("server", "console")
+                .getFieldValue("server", Bukkit.getServer());
+        this.reader = (ConsoleReader) ReflectionProvider.DEFAULT.createNMSReflect("MinecraftServer", "MinecraftServer")
+                .searchField("console", "reader")
+                .getFieldValue("console", mcServer);
+        this.writer = reader.getOutput();
+        this.ansiSupported = reader.getTerminal().isAnsiSupported();
+        this.jLine = Main.useJline;
+
+        setDefaults();
+    }
+
+    @Override
+    public ILogger close() {
+        throw new UnsupportedOperationException("BukkitLogger cannot be closed");
+    }
+
+    @Override
+    public LogTypeMap getTypeMap() {
+        return typeMap;
+    }
+
+    @Override
+    public ILogger setCustom(BiConsumer<Boolean, String> biConsumer) {
+        throw new UnsupportedOperationException("BukkitLogger doesn't support setCustom");
+    }
+
+    @Override
+    public BiConsumer<Boolean, String> getCustom() {
+        throw new UnsupportedOperationException("BukkitLogger doesn't support getCustom");
+    }
+
+    @Override
+    public ILogger setType(LogType logType) {
+        typeMap.override(logType);
+        return this;
+    }
+
+    @Override
+    public LogType getType(String s) {
+        return typeMap.tryGet(type -> type.getId().equals(s)).orElse(BukkitLogType.DEFAULT);
+    }
+
+    private void setDefaults() {
+        setType(new BukkitLogType("debug", ChatColor.GRAY));
+        setType(new BukkitLogType("info", ChatColor.AQUA));
+        setType(new BukkitLogType("warning", ChatColor.GOLD));
+        setType(new BukkitLogType("error", ChatColor.RED));
+    }
+
+    public String rgb(int red, int green, int blue) {
```

Comment:
Use a constant or utility class instead, maybe just the LoggingColors.java class?